src/gosling-track/gosling-track.ts

```diff
@@ -542,67 +542,84 @@ function GoslingTrack(HGC: import('@higlass/types').HGC, ...args: any[]): any {
                         // it makes sense only when the y-axis is being used for a genomic field
-                        this.yTiles = tileProxy.calculateTilesFromResolution(
-                            sortedResolutions[this.zoomLevel],
+                        yTiles = tileProxy.calculateTilesFromResolution(
+                            sortedResolutions[zoomLevel],
                             this._yScale,
                             this.tilesetInfo.min_pos[0],
                             this.tilesetInfo.max_pos[0]
                         );
                     }
 
-                    const tiles = this.tilesToId(this.xTiles, this.yTiles, this.zoomLevel);
+                    const tiles = this.tilesToId(xTiles, yTiles, zoomLevel);
+
                     this.setVisibleTiles(tiles);
                 } else {
-                    this.xTiles = tileProxy.calculateTiles(
-                        this.zoomLevel,
+                    const xTiles = tileProxy.calculateTiles(
+                        zoomLevel,
                         this.relevantScale(),
                         this.tilesetInfo.min_pos[0],
                         this.tilesetInfo.max_pos[0],
                         this.tilesetInfo.max_zoom,
                         this.tilesetInfo.max_width
                     );
 
+                    let yTiles: number[] | undefined;
                     if (Is2DTrack(resolveSuperposedTracks(this.options.spec)[0])) {
                         // it makes sense only when the y-axis is being used for a genomic field
-                        this.yTiles = tileProxy.calculateTiles(
-                            this.zoomLevel,
+                        yTiles = tileProxy.calculateTiles(
+                            zoomLevel,
                             this._yScale,
                             this.tilesetInfo.min_pos[1],
                             this.tilesetInfo.max_pos[1],
                             this.tilesetInfo.max_zoom,
-                            this.tilesetInfo.max_width1 || this.tilesetInfo.max_width
+                            // @ts-expect-error what is max_width1?
+                            this.tilesetInfo.max_width1 ?? this.tilesetInfo.max_width
                         );
                     }
 
-                    const tiles = this.tilesToId(this.xTiles, this.yTiles, this.zoomLevel);
+                    const tiles = this.tilesToId(xTiles, yTiles, zoomLevel);
                     this.setVisibleTiles(tiles);
                 }
             }
         }
 
+        private get binsPerTile() {
+            let maybeValue: number | undefined;
+            if (this.tilesetInfo) {
+                maybeValue =
+                    'bins_per_dimension' in this.tilesetInfo
+                        ? this.tilesetInfo.bins_per_dimension
+                        : this.tilesetInfo.tile_size;
+            }
+            return maybeValue ?? 256;
+        }
```

Comment:
This logic was repeated all over the place. 

Moving it to a private class property means that we can modify the logic in the future and not worry about needing to keep everything in sync.